misc/generic_editor.py

```diff
@@ -236,30 +208,57 @@ def word_prev(m):
-        # META
-        "sage": Key("cmd-s"),
-        "dizzle": Key("cmd-z"),
-        "rizzle": Key("cmd-shift-z"),
+        # meta
+        "(save it | sage)": Key("cmd-s"),
+        "(undo it | dizzle)": Key("cmd-z"),
+        "(redo it | rizzle)": Key("cmd-shift-z"),
+        # clipboard
+        "([clip] cut | snatch)": Key("cmd-x"),
+        "([clip] copy | stoosh)": Key("cmd-c"),
+        "([clip] paste | spark)": Key("cmd-v"),
         # MOTIONS
-        "spring" + optional_numerals: jump_to_eol_and(jump_to_beginning_of_text),
-        "dear" + optional_numerals: jump_to_eol_and(lambda: None),
-        "smear" + optional_numerals: jump_to_eol_and(jump_to_nearly_end_of_line),
-        # 'sprinkoon' + numerals: jump_to_eol_and(lambda: press('enter')),
-        "shockey": Key("ctrl-a cmd-left enter up"),
-        "shockoon": Key("cmd-right enter"),
-        "jolt": Key("ctrl-a cmd-left shift-down cmd-c down cmd-v"),  # duplicate line
-        # DELETING
-        "snipple": Key("shift-cmd-left delete"),
-        "snipper": Key("shift-cmd-right delete"),
-        "shackle": Key("cmd-right shift-cmd-left"),
-        "snipline" + optional_numerals: jump_to_bol_and(snipline),
-        # SELECTING
-        "sprinkle" + optional_numerals: select_line,
-        "crew <dgndictation>": select_text_to_right_of_cursor,
-        "trail <dgndictation>": select_text_to_left_of_cursor,
-        "shift home": Key("shift-home"),
-        "wordneck" + optional_numerals: word_neck,
-        "wordprev" + optional_numerals: word_prev,
-        "word this": [Key("alt-right"), Key("shift-alt-left")],
+        "(go word left | fame | peg)": Key("alt-left"),
+        "(go word right | fish | fran)": Key("alt-right"),
+        "(go after line end | derek)": Key("cmd-right space"),
+        "(go line start | spring | lefty) {generic_editor.n}*": select_line_and_press(
+            ("ctrl-a", "cmd-left")
+        ),
+        "(go line end | dear | ricky) {generic_editor.n}*": select_line_and_press(
+            ("cmd-right")
+        ),
+        "(go before line end | smear) {generic_editor.n}*": select_line_and_press(
+            ("cmd-right", "left")
+        ),
+        "(new line below | sprinkoon | slap) {generic_editor.n}*": select_line_and_press(
+            ("cmd-right", "enter")
+        ),
+        "(new line above | shockey | shocker)": Key("ctrl-a cmd-left enter up"),
+        "(duplicate line | jolt)": Key("ctrl-a cmd-left shift-down cmd-c down cmd-v"),
+        # deleting
+        "(delete around this | slurp)": Key("backspace delete"),
+        "(delete line left | snip left | snipple)": Key("shift-cmd-left delete"),
+        "(delete line right | snip right | snipper)": Key("shift-cmd-right delete"),
+        "(delete [this] line | snipline ) {generic_editor.n}*": select_line_and_press(
+            ("shift-cmd-right", "delete", "delete", "ctrl-a", "cmd-left")
+        ),
+        "(delete word left | trough | steffi | carmex)": Key("alt-backspace"),
```

Comment:
I have kept the commands as aliases where multiple commands existed for the same action. Do we want to do that? Maybe just preserving just a single one-word alternative is preferable? Surely these had different meanings anyway in Voicecode? But I don't know which is the 'right' one for this command.